### PR TITLE
[rails__actioncable] Add readonly props, update types

### PR DESCRIPTION
**Pull request by martin-badin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

**Comment by martin-badin** [comment]
@pesoklp13 Hi. Please review. Thanks :)

**Review by pesoklp13 — APPROVED** [review]
I like generics thou

**Comment by OscarBarrett** [comment]
One note - `createConsumer` is currently typed as `(url: string): typeof Consumer` however [`url` has a guaranteed default](https://github.com/rails/rails/blob/main/actioncable/app/javascript/action_cable/index.js#L22), so should be made optional.

**Comment by martin-badin** [comment]
> One note - `createConsumer` is currently typed as `(url: string): typeof Consumer` however [`url` has a guaranteed default](https://github.com/rails/rails/blob/main/actioncable/app/javascript/action_cable/index.js#L22), so should be made optional.

thank, updated

**Comment by martin-badin** [comment]
Ready to merge

**Comment by martin-badin** [comment]
@OscarBarrett Could you review it?

**Review by OscarBarrett — APPROVED** [review]

**Comment by martin-badin** [comment]
@pesoklp13 please review again. Thanks

**Comment by martin-badin** [comment]
Ready to merge

File: types/rails__actioncable/index.d.ts [hunk 1]
```diff
@@ -73,8 +101,12 @@ export class Connection {
 /**
  * @see https://github.com/rails/rails/blob/main/actioncable/app/javascript/action_cable/connection_monitor.js
  */
-export class ConnectionMonitor {
-    constructor(connection: Connection);
+export class ConnectionMonitor<C = Consumer> {
```
Comment (ratiofu) [inline types/rails__actioncable/index.d.ts:104]: Hello @martin-badin 👋 Shouldn't this be `ConnectionMonitor<C = Connection>`? See the constructor ↓ Happy to provide a PR if needed.

**Pull request merged**